rmgpy/rmgobject.pyx

```diff
@@ -133,3 +134,52 @@ cpdef expand_to_dictionaries(obj):
             return expand_to_dictionaries(obj.as_dict())
         else:
             return obj
+
+cpdef recursive_make_object(obj, class_dictionary, make_final_object=True):
```

Comment:
Do we ever call this with `make_final_object=False`?

Reply:
Yes, this gets called with `make_final_object=False` in the RMGObject `make_object` method. This is because the `make_object` method passes the dictionary as keyword arguments into the __init__ method. By doing this, it initializes the object in place.

If you call the function with `make_final_object=True` then the function returns a new instance of the object (i.e. not in place). This mean that if you read in a dictionary from a yaml file, you can simply call this function on the dictionary and it will return the object for you (you do not need to create an "empty" instance of the object first.

However, if you already have an empty instance or want to reinitialize the object, you just need to call the `make_object` method of the object.